llvm/lib/Target/AMDGPU/AMDGPUCodeGenPrepare.cpp

```diff
@@ -330,6 +330,9 @@ class AMDGPUCodeGenPrepareImpl
   bool visitBitreverseIntrinsicInst(IntrinsicInst &I);
   bool visitMinNum(IntrinsicInst &I);
   bool visitSqrt(IntrinsicInst &I);
+
+  bool visitAllocaInst(AllocaInst &I);
```

Comment:
We might want to run this earlier to make it more useful, although I'm a bit skeptical about how practically useful it'll be in the end. This is more like a safeguard for irregular code paths. Normal code definitely wouldn't emit this kind of alloca.